src/context/search-context.js

```diff
@@ -111,9 +111,11 @@ export function useSearch() {
   return [useSearchState(), useSearchDispatch()]
 }
 
-async function queryIndex(index, query) {
+async function queryIndex(index, query, distinct = false) {
```

Comment:
Related to my first comment about the `attributeForDisctinct`: well done, what I understood from the docs was "either at indexing time or query time" but the "once attributeForDistinct is set, you can enable distinct" line just proved me wrong!